crates/gritmodule/src/testing.rs

```diff
@@ -32,6 +33,29 @@ pub fn collect_testable_patterns(
     testable_patterns
 }
 
+fn map_file_pattern_to_test_info(pattern: &mut ModuleGritPattern) -> GritPatternTestInfo {
+    let samples = pattern.config.samples.take();
+    GritPatternTestInfo {
+        body: pattern.config.body.clone().unwrap(),
+        config: GritPatternTestConfig {
+            path: Some(pattern.config.path.clone()),
+            samples,
+        },
+        local_name: Some(pattern.local_name.clone()),
+    }
+}
+
+pub fn get_grit_pattern_test_info(
+    mut patterns: Vec<ModuleGritPattern>,
+) -> Vec<GritPatternTestInfo> {
+    let testable_patterns: Vec<GritPatternTestInfo> = patterns
+        .iter_mut()
```

Comment:
Use into_iter and avoid needing to clone.

Reply:
done